lib/pinecone.ex

```diff
@@ -105,40 +108,72 @@ defmodule Pinecone do
       level configuration. Defaults to `nil`
   """
   @spec create_index(index_name :: String.t(), opts :: keyword()) ::
-          success_type(String.t()) | error_type()
+          success_type(map) | error_type()
   def create_index(index_name, opts \\ []) do
     opts =
       Keyword.validate!(opts, [
         :config,
         dimension: 384,
         metric: :euclidean,
-        pods: 1,
-        replicas: 1,
-        shards: 1,
-        pod_type: {:p1, :x1},
-        metadata: []
+        spec: []
       ])
 
+    # TODO: validate metadata
     validate!("index_name", index_name, :binary)
     validate!("dimension", opts[:dimension], :non_negative_integer)
-    validate!("pods", opts[:pods], :non_negative_integer)
-    validate!("replicas", opts[:replicas], :non_negative_integer)
-    validate!("shards", opts[:shards], :non_negative_integer)
     validate!("metric", opts[:metric], :one_of, @valid_metric_types)
-    validate!("pod_type", opts[:pod_type], :one_of, @valid_pods)
-    # TODO: validate metadata
+
+    spec_params = validate_create_index_opts(opts[:spec])
 
     body = %{
       "name" => index_name,
       "dimension" => opts[:dimension],
       "metric" => Atom.to_string(opts[:metric]),
-      "pods" => opts[:pods],
-      "replicas" => opts[:replicas],
-      "shards" => opts[:shards],
-      "pod_type" => to_pod_type(opts[:pod_type])
+      "spec" => spec_params
     }
 
-    post(:indices, "databases", body, opts[:config])
+    post(:indices, "", body, opts[:config])
+  end
+
+  defp validate_create_index_opts(spec_opts) do
+    case spec_opts do
+      [serverless: serverless_opts] ->
+        serverless_opts = Keyword.validate!(serverless_opts,
+          cloud: "aws",
+          region: "us-west-2"
+        )
+
+        validate!("cloud", serverless_opts[:cloud], :one_of, @valid_clouds)
+        validate!("region", serverless_opts[:region], :binary)
+
+        %{"serverless" => %{
+          "cloud" => serverless_opts[:cloud],
+          "region" => serverless_opts[:region]
+        }}
+
+      [pod: pod_opts] ->
+        pod_opts = Keyword.validate!(pod_opts,
+          pods: 1,
+          replicas: 1,
+          shards: 1,
+          pod_type: {:p1, :x1},
+          metadata: []
+        )
+
+        validate!("pods", pod_opts[:pods], :non_negative_integer)
+        validate!("replicas", pod_opts[:replicas], :non_negative_integer)
+        validate!("shards", pod_opts[:shards], :non_negative_integer)
+        validate!("pod_type", pod_opts[:pod_type], :one_of, @valid_pods)
+
+        %{"pod" => %{
+          "pods" => pod_opts[:pods],
+          "replicas" => pod_opts[:replicas],
+          "shards" => pod_opts[:shards],
+          "pod_type" => to_pod_type(pod_opts[:pod_type])
+        }}
```

Comment:
I think you want to run the formatter here :)

Reply:
Ah yes, I'm notoriously bad with running formatter 😂  I'll do that